api/src/main/java/jakarta/enterprise/lang/model/AnnotationMemberValue.java

```diff
@@ -0,0 +1,262 @@
+     */
+    default boolean isArray() {
+        return kind() == Kind.ARRAY;
+    }
+
+    /**
+     * @return Returns {@code true} if the kind is an {@link java.lang.annotation.Annotation}, {@code false} otherwise.
+     */
+    default boolean isNestedAnnotation() {
+        return kind() == Kind.NESTED_ANNOTATION;
+    }
+
+    /**
+     * Return the value as a boolean.
+     * @return The boolean value
+     */
+    boolean asBoolean();
+
+    /**
+     * Return the value as a byte.
+     * @return The byte value
+     * @throws NumberFormatException if the value cannot be represented as a byte.
+     */
+    byte asByte();
+
+    /**
+     * Return the value as a short.
+     * @return The short value
+     * @throws NumberFormatException if the value cannot be represented as a short.
+     */
+    short asShort();
+
+    /**
+     * Return the value as an int.
+     * @return The int value
+     * @throws java.lang.IllegalStateException if the value cannot be represented as an int.
+     */
+    int asInt();
+
+    /**
+     * Return the value as a long.
+     * @return The long value
+     * @throws java.lang.IllegalStateException if the value cannot be represented as a long.
+     */
+    long asLong();
+
+    /**
+     * Return the value as a float.
+     * @return The float value
+     * @throws java.lang.IllegalStateException if the value cannot be represented as a float.
+     */
+    float asFloat();
+
+    /**
+     * Return the value as a double.
+     * @return The double value
+     * @throws java.lang.IllegalStateException if the value cannot be represented as a double.
+     */
+    double asDouble();
+
+    /**
+     * Return the value as a char.
+     * @return The char value
+     * @throws java.lang.IllegalStateException if the value cannot be represented as a double.
+     */
+    char asChar();
+
+    /**
+     * Return the value as a string.
+     * @return A string representing the value. Never {@code null}.
+     */
+    String asString();
+
+    /**
+     * Return the value to an enum instance.
+     * @param enumType The enum type
+     * @param <E> The enum generic type
+     * @return The enum instance
+     * @throws java.lang.IllegalStateException if the enum value cannot be established from the given argument
+     */
+    <E extends Enum<E>> E asEnum(Class<E> enumType);
+
+    /**
+     * Return the enum type that an enum value represents.
+     *
+     * @return A {@link jakarta.enterprise.lang.model.declarations.ClassInfo} representing the type of an enum value.
+     */
+    ClassInfo<?> asEnumClass();
+
+    /**
+     * Return the type of the value. Valid types include {@link jakarta.enterprise.lang.model.types.PrimitiveType}, {@link jakarta.enterprise.lang.model.types.VoidType} and {@link jakarta.enterprise.lang.model.types.ClassType}.
+     *
+     * @return The {@link jakarta.enterprise.lang.model.types.Type} of the value.
+     */
+    Type asType(); // can be a VoidType, PrimitiveType or ClassType
```

Comment:
I originally forgot it can also be an array, but I can amend that in my follow-up PR.